### PR TITLE
MARS-10: Display NASA's astronomy picture of the day

### PR DESCRIPTION
**Pull request by leedoughty**
**Changes**
- Display NASA's astronomy picture of the day on the homepage
- Display photo explanation from NASA api
Screenshot 2020-11-25 at 09 01 05

### CODE REVIEW

File: public/displayPhoto.js [hunk 1]
```diff
@@ -0,0 +1,10 @@
+function displayPhoto() {
+  fetch("http://localhost:3000/photo")
```
Comment (Kenneth-Siu) [inline public/displayPhoto.js]: If we were to ever deploy this to a server, we'd want this to be a relative URL

File: middleware/getPhotoOfTheDay.js [hunk 2]
```diff
@@ -0,0 +1,16 @@
+const fetch = require("node-fetch");
+const API_KEY = require("../apiKey");
+const BASE_URL = "https://api.nasa.gov/planetary/apod";
+
+const getPhotoOfTheDay = async (request, response, next) => {
+  const url = `${BASE_URL}?api_key=${API_KEY}`;
+  try {
+    const apodResponse = await fetch(url);
```
Comment (Kenneth-Siu) [inline middleware/getPhotoOfTheDay.js]: What happens if NASA's API is down or if there's a problem with our code somewhere?

Does our server return a response?

We'd ideally like our endpoint to give some sort of error code. Let's start by a blanket decision that if there's a problem, our endpoint should return a response with HTTP status code 500.

**Review by Kenneth-Siu — COMMENTED** [review]
We've reached the interesting bit about how our APIs should handle internal issues, and what it then returns to the user in its response! Feel free to grab me if you're not sure where to start

File: middleware/getPhotoOfTheDay.js (hunk 2 above)
Comment (leedoughty) [reply middleware/getPhotoOfTheDay.js]: I've setup some error handling, let me know what you think :)

File: middleware/getPhotoOfTheDay.js [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+  } else {
+    throw new Error("error");
+  }
+};
+
+const getPhotoOfTheDay = async (request, response, next) => {
+  const url = `${BASE_URL}?api_key=${API_KEY}`;
+  try {
+    const apodResponse = await fetch(url).then(checkStatus);
+    const apodJson = await apodResponse.json();
```
Comment (Kenneth-Siu) [inline middleware/getPhotoOfTheDay.js:17]: I'd be tempted to put together these two lines into something like `await fetch(url).then(checkStatus).json()` but this is a very minor thing that has a lot of personal preference attached.

File: package.json [hunk 4]
```diff
@@ -18,7 +18,9 @@
   },
   "homepage": "https://github.com/leedoughty/mars-rover#readme",
   "dependencies": {
-    "express": "^4.17.1"
+    "axios": "^0.21.0",
```
Comment (Kenneth-Siu) [inline package.json]: Whoops, don't think this is being used any more :)

File: public/displayPhoto.js [hunk 5]
```diff
@@ -0,0 +1,25 @@
...
+      const photoOfTheDay = document.getElementById("photo-of-the-day");
+      const photoExplanation = document.getElementById("photo-explanation");
+      photoOfTheDay.setAttribute("src", `${data.url}`);
+      photoOfTheDay.setAttribute("alt", `${data.title}`);
+      photoExplanation.textContent = data.explanation;
+    })
+    .catch(() => {
+      const photoOfTheDay = document.getElementById("photo-of-the-day");
+      photoOfTheDay.setAttribute("src", "mars-rover.jpg");
+      photoOfTheDay.setAttribute("alt", "The Mars Rover");
```
Comment (Kenneth-Siu) [inline public/displayPhoto.js:23]: Nice, showing a backup image if it doesn't work. 👌

**Review by Kenneth-Siu — COMMENTED** [review]
Excellent, looks really good!

A couple of teeny-tiny things, one of which is optional. I'm happy for you to go ahead and merge without waiting for me to approve it once you've sorted them out.

File: middleware/getPhotoOfTheDay.js (hunk 3 above)
Comment (leedoughty) [reply middleware/getPhotoOfTheDay.js:17]: Ah I did consider this, although I slightly lean more towards having these both on the two lines as individual constants as I think this makes the readability a bit more clearer.

File: package.json (hunk 4 above)
Comment (leedoughty) [reply package.json]: Ah good spot! 😄 I started looking into using axios originally, but wanted to try out node-fetch instead (as I hadn't used it before), I will remove axios as a dependency now

**Pull request merged**